src/sentry/search/django/backend.py

```diff
@@ -250,6 +250,69 @@ def query(self, project, tags=None, environment=None, sort_by='date', limit=100,
         else:
             retention_window_start = None
 
+        # TODO: some flag to decide (per user/project?) whether snuba should be used for search
+        snuba = False
+
+        if snuba:
+            if tags or \
+                    environment is not None or \
+                    any(key in parameters for key in ('date_from', 'date_to')):
+                # if we're in this branch then Snuba can help us filter down the eligible group ids!
+
+                # TODO: move existing logic from `get_group_ids_for_search_filter` to filter this query on tags
+                # TODO: translate to snuba.utils.query obviously :)
+                # TODO: only pass the necessary filters ('environment', 'date_from',
+                #       'date_to' are optional)
+                # TODO: snuba's `get_group_ids_for_search_filter` and `get_event_tag_qs`
+                #       are no longer necessary
+                hashes = [
```

Comment:
this logic (fetching hashes and group_ids) should probably be combined with https://github.com/getsentry/sentry/blob/c65803c884557690a208595cf09c8b9801beedf8/src/sentry/utils/snuba.py#L146

Reply:
If you aren't doing anything too fancy, `get_project_issues` might already be doing the correct magic for you when you call `snuba.query()`